### ISSUE TITLE
[NLL] Many errors generated instead of one with many notes

### ISSUE THREAD

**Issue by marmeladema** [rust-lang/rust#96278]
In tests like:
* [src/test/ui/regions/regions-addr-of-upvar-self.rs](https://github.com/rust-lang/rust/tree/master/src/test/ui/regions/regions-addr-of-upvar-self.rs)
* [src/test/ui/regions/regions-close-object-into-object-4.rs](https://github.com/rust-lang/rust/tree/master/src/test/ui/regions/regions-close-object-into-object-4.rs)

Under NLL we generate many errors for almost the same span instead of one error with many notes under migrate mode, which is more readable and easier to understand.

We should fix this.

@rustbot label +A-NLL +NLL-diagnostics

**Comment by jackh726** [comment]
Any way this is fixed by #96236?

**Comment by marmeladema** [comment]
No, I rebased over current master and blessed all the tests and nothing changed.
